arm/Microsoft.Compute/disks/deploy.bicep

```
@@ -0,0 +1,188 @@
+param lock string = 'NotSpecified'
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'')
+param roleAssignments array = []
+
+@description('Optional. Tags of the availability set resource.')
+param tags object = {}
+
+@description('Optional. Customer Usage Attribution ID (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource disk 'Microsoft.Compute/disks@2021-08-01' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: sku
+  }
+  properties: {
+    burstingEnabled: burstingEnabled
+    completionPercent: completionPercent
+    creationData: {
+      createOption: createOption
+      imageReference: createOption != 'FromImage' ? null : {
+        id: imageReferenceId
+      }
+      logicalSectorSize: contains(sku, 'Ultra') ? logicalSectorSize : null
+      securityDataUri: createOption == 'ImportSecure' ? securityDataUri : null
+      sourceResourceId: createOption == 'Copy' ? sourceResourceId : null
+      sourceUri: createOption == 'Import' ? sourceUri : null
+      storageAccountId: createOption == 'Import' ? storageAccountId : null
+      uploadSizeBytes: createOption == 'Upload' ? uploadSizeBytes : null
+    }
+    diskIOPSReadWrite: contains(sku, 'Ultra') ? diskIOPSReadWrite : null
+    diskMBpsReadWrite: contains(sku, 'Ultra') ? diskMBpsReadWrite : null
+    diskSizeGB: createOption == 'Empty' ? diskSizeGB : null
+    hyperVGeneration: empty(osType) ? null : hyperVGeneration
+    maxShares: maxShares
+    networkAccessPolicy: networkAccessPolicy
+    osType: empty(osType) ? null : osType
+    publicNetworkAccess: publicNetworkAccess
+    supportedCapabilities: empty(osType) ? {} : {
+      acceleratedNetwork: acceleratedNetwork
+    }
+  }
+}
+
+resource disk_lock 'Microsoft.Authorization/locks@2017-04-01' = if (lock != 'NotSpecified') {
+  name: '${disk.name}-${lock}-lock'
+  properties: {
+    level: lock
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: disk
+}
+
+module disk_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${uniqueString(deployment().name, location)}-AvSet-Rbac-${index}'
+  params: {
+    principalIds: roleAssignment.principalIds
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    resourceId: disk.id
+  }
+}]
+
+@description('The resource group the  disk was deployed into')
+output diskResourceGroup string = resourceGroup().name
```

In line with the latest output approach as per issue #623 
```suggestion
@description('The resource group the disk was deployed into')
output resourceGroupName string = resourceGroup().name
```